nodes/src/nodes/tool_git/git_repo.py

```
@@ -0,0 +1,972 @@
+        if analysis & _GIT_MERGE_ANALYSIS_FASTFORWARD:
+            repo.checkout_tree(remote_commit)
+            repo.head.set_target(remote_commit.id)
+            return {**fetch_result, 'merge': 'fast_forwarded', 'sha': str(remote_commit.id)[:8]}
+
+        raise GitError('Cannot fast-forward. The branch has diverged. Merge manually or rebase before pulling.')
+
+    def push(
+        self,
+        remote: str = 'origin',
+        branch: Optional[str] = None,
+        force: bool = False,
```

Ruff (0.15.12)

[warning] Boolean-typed positional argument in function definition

(FBT001)